batch/batch/worker/worker.py

```diff
@@ -1674,11 +1699,15 @@ async def run(self):
                 self.state = 'failed'
                 self.error = traceback.format_exc()
                 await self.cleanup()
+            except JobDeletedError:
+                self.state = 'cancelled'
+                await self.cleanup()
```

Comment:
Hmm. This is interesting. We don't cleanup if we're cancelled in the JVMJob but we do cleanup if we're cancelled in the DockerJob. Hmm. We only get cancelled if the worker is getting shutdown, so it does seem wise to just go down ASAP without cleaning anything up. OK, let's not deal with this now, but we should revisit this if refactor the worker.